src/platform/vita/vita_keyboard.c

```diff
@@ -109,27 +110,20 @@ const uint8_t *vita_keyboard_get(const uint8_t *initial_text, int max_length)
 
     if (ime_init_apputils == 0) {
         sceAppUtilInit(&(SceAppUtilInitParam){}, &(SceAppUtilBootParam){});
-        sceCommonDialogSetConfigParam(&(SceCommonDialogConfigParam){});
         ime_init_apputils = 1;
     }
     init_ime_dialog(initial_text, max_length, SCE_IME_TYPE_BASIC_LATIN, 0);
     int done = 0;
     while (!done) {
-        vita2d_start_drawing();
-        vita2d_clear_screen();
-
+        platform_screen_clear();
         done = 1;
-
         int ime_result = update_ime_dialog();
         if (ime_result == IME_DIALOG_RESULT_FINISHED) {
             encoding_from_utf8(ime_text_utf8, final_text, max_length);
         } else if (ime_result != IME_DIALOG_RESULT_CANCELED) {
             done = 0;
         }
-        vita2d_end_drawing();
-        vita2d_common_dialog_update();
-        vita2d_swap_buffers();
-        sceDisplayWaitVblankStart();
+        platform_screen_render(0);
```

Comment:
Just wondering: what's the reason for not wanting to redraw the screen at this point?

Reply:
The normal redraw gets the drawing from our texture which is controlled by `graphics_canvas`. When drawing the virtual keyboard, we don't want/need the texture from there but rather the internal rendering from vita's virtual keyboard.  So we only need to call renderer but not to update the texture.